### PR TITLE
MF-667: Patient List Detail View

### PR DESCRIPTION
**Pull request by vasharma05**
### What does this PR include?

This PR contains the page for a specific patient-list at the url `/patient-list/:patientListUuid/`

Design is similar to: https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/60d5974313f68b0569f2c55b

### CODE REVIEW

File: packages/esm-patient-list-app/src/index.ts [hunk 1]
```diff
@@ -31,15 +31,15 @@ function setupOpenMRS() {
     },
     {
       path: `${spaBasePath}/:view?`,
-      title: ([x]) => `${x} Dashboard`,
+      title: ([x]) => 'Dashboard',
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/index.ts:34]: Is this accurate or should it be "Members"? I guess the `:view?` may also be `:view` (i.e., non-optional, this has to be given)

File: packages/esm-patient-list-app/src/patient-list/patient-list-detail.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+@import "../style.scss";
+
+.patientListBanner {
+  border-top: 1px solid #e0e0e0;
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/patient-list/patient-list-detail.scss:4]: -> color is `$ui-03`

File: packages/esm-patient-list-app/src/patient-list/patient-list-detail.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  display: flex;
+  justify-content: space-between;
+}
+
+.leftBannerSection {
+  padding: 1rem;
+}
+
+.secondaryText {
+  color: #525252;
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/patient-list/patient-list-detail.scss:14]: -> color is `$text-02`

File: packages/esm-patient-list-app/src/style.scss [hunk 4]
```diff
@@ -9,6 +9,18 @@
...
+
+.bodyShort02 {
+  @include carbon--type-style("body-long-02");
+}
+
+.label01 {
+  @include carbon--type-style("label-01");
+}
+
 .modal {
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/style.scss:24]: Where is this `modal` used? I think it should be removed.

File: packages/esm-patient-list-app/src/ui-components/overflow-menu.component.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        height: 'auto',
+      }}
+      ref={wrapperRef}>
+      <button
+        className={`bx--overflow-menu__trigger ${showMenu && 'bx--overflow-menu--open'}`}
+        aria-haspopup="true"
+        aria-expanded={showMenu}
+        id="custom-actions-overflow-menu-trigger"
+        aria-controls="custom-actions-overflow-menu"
+        onClick={() => toggleShowMenu(!showMenu)}
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/ui-components/overflow-menu.component.tsx:44]: Put that in a `useCallback` like this:

```js
const toggle = useCallback(() => toggleShowMenu(state => !state), []);
```

Please rename `toggleShowMenu` to `setShowMenu`.

File: packages/esm-patient-list-app/src/ui-components/overflow-menu.component.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        style={{
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/ui-components/overflow-menu.component.tsx:45]: Put that in the sass file and reference is via a class name.

File: packages/esm-patient-list-app/src/ui-components/overflow-menu.component.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        {menuTitle}
+      </button>
+      <div
+        className="bx--overflow-menu-options bx--overflow-menu--flip"
+        tabIndex={0}
+        data-floating-menu-direction="bottom"
+        role="menu"
+        aria-labelledby="custom-actions-overflow-menu-trigger"
+        id="custom-actions-overflow-menu"
+        style={{
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/ui-components/overflow-menu.component.tsx:61]: -> CSS

File: packages/esm-patient-list-app/src/patient-list/patient-list-details.component.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  return (
+    <div className="omrs-main-content">
+      <ExtensionSlot extensionSlotName="breadcrumbs-slot" />
+      <div className={styles.patientListBanner}>
+        <div className={styles.leftBannerSection}>
+          <h5 className={styles.bodyShort02}>
+            {t("A list of patients that haven't been seen for 30 days, since their missed appointment")}
+          </h5>
+          <div className={`${styles.secondaryText} ${styles.bodyShort01}`} style={{ marginTop: '0.5rem' }}>
+            <span>128 {t('patients', 'patients')}</span> · <span className={styles.label01}>Last Updated:</span> 12 /
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/patient-list/patient-list-details.component.tsx:21]: Why is last updated and the number of patients being hardcoded?

File: packages/esm-patient-list-app/src/patient-list/patient-list-details.component.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ExtensionSlot } from '@openmrs/esm-framework';
+import CustomOverflowMenu from '../ui-components/overflow-menu.component';
+import OverflowMenuVertical16 from '@carbon/icons-react/es/overflow-menu--vertical/16';
+import PatientListDataTable from '../patient-table/patient-table.component';
+import styles from './patient-list-detail.scss';
+
+const PatientListDetails: React.FC<{}> = () => {
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/patient-list/patient-list-details.component.tsx:9]: It should be `React.FC<PatientListDetailsProps>` where the `PatientListDetailsProps` are

```
interface PatientListDetailsProps extends RouteComponentProps {}
```

to reflect it coming from the React router. Also note that the params should ideally contain the `patientListUuid`

File: packages/esm-patient-list-app/src/patient-list/patient-list-details.component.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  const { t } = useTranslation();
+
+  return (
```
Comment (FlorianRappl) [inline packages/esm-patient-list-app/src/patient-list/patient-list-details.component.tsx:12]: Missing the whole details list (obtained by its UUID `patientListUuid`).

**Review by FlorianRappl — COMMENTED** [review]
I think we are on a good way here, but right now it misses the actual list.

**Comment by vasharma05** [comment]
image

**Comment by FlorianRappl** [comment]
Are you taking care of the conflicts @vasharma05 ?

**Comment by vasharma05** [comment]
Hi @FlorianRappl! Yes I am handling the conflicts, some changes are still pending which are to be done and pushed by today possibly.

**Comment by FlorianRappl** [comment]
Hi @vasharma05 any update on this one?

**Pull request closed without merging**